src/Service/S3/src/Input/PutPublicAccessBlockRequest.php

```diff
@@ -0,0 +1,224 @@
+            throw new InvalidArgument(\sprintf('Missing parameter "Bucket" for "%s". The value cannot be null.', __CLASS__));
+        }
+        $uri['Bucket'] = $v;
+        $uriString = '/' . rawurlencode($uri['Bucket']) . '?publicAccessBlock';
+
+        // Prepare Body
+        $document = new \DOMDocument('1.0', 'UTF-8');
+        $document->formatOutput = false;
+        $this->requestBody($document, $document);
+        $body = $document->hasChildNodes() ? $document->saveXML() : '';
+
+        // Headers MUST be prepared AFTER the body, so that Content-MD5 can be calculated.
+        $headers = ['content-type' => 'application/xml'];
+        if (null !== $this->contentMd5) {
+            $headers['Content-MD5'] = $this->contentMd5;
+        }
+        if (null !== $this->checksumAlgorithm) {
+            if (!ChecksumAlgorithm::exists($this->checksumAlgorithm)) {
+                throw new InvalidArgument(\sprintf('Invalid parameter "ChecksumAlgorithm" for "%s". The value "%s" is not a valid "ChecksumAlgorithm".', __CLASS__, $this->checksumAlgorithm));
+            }
+            $headers['x-amz-sdk-checksum-algorithm'] = $this->checksumAlgorithm;
+        }
+        if (null !== $this->expectedBucketOwner) {
+            $headers['x-amz-expected-bucket-owner'] = $this->expectedBucketOwner;
+        }
+
+        // Return the Request
+        return new Request('PUT', $uriString, $query, $headers, StreamFactory::create($body));
+    }
+
+    public function setBucket(?string $value): self
+    {
+        $this->bucket = $value;
+
+        return $this;
+    }
+
+    /**
+     * @param ChecksumAlgorithm::*|null $value
+     */
+    public function setChecksumAlgorithm(?string $value): self
+    {
+        $this->checksumAlgorithm = $value;
+
+        return $this;
+    }
+
+    public function setContentMd5(?string $value): self
+    {
+        $this->contentMd5 = $value;
+
+        return $this;
+    }
+
+    public function setExpectedBucketOwner(?string $value): self
+    {
+        $this->expectedBucketOwner = $value;
+
+        return $this;
+    }
+
+    public function setPublicAccessBlockConfiguration(?PublicAccessBlockConfiguration $value): self
+    {
+        $this->publicAccessBlockConfiguration = $value;
+
+        return $this;
+    }
+
+    private function requestBody(\DOMNode $node, \DOMDocument $document): void
+    {
+        if (null === $v = $this->publicAccessBlockConfiguration) {
+            throw new InvalidArgument(\sprintf('Missing parameter "PublicAccessBlockConfiguration" for "%s". The value cannot be null.', __CLASS__));
+        }
+
+        $node->appendChild($child = $document->createElement('PublicAccessBlockConfiguration'));
+        $child->setAttribute('xmlns', 'http://s3.amazonaws.com/doc/2006-03-01/');
+        $v->requestBody($child, $document);
+
+        if (null === $this->contentMd5) {
+            $this->contentMd5 = \hash('md5', $document->saveXML($child));
+        }
```

Comment:
@stof I assume that this functionality should be included, to mimic the official SDK behavior?

Reply:
It looks like the one failing check is because of this change. It seems that similar parameters are _not_ automatically resolved by async-aws, so I will remove it.